session/objecthandling.go

```diff
@@ -0,0 +1,741 @@
+package session
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/qlik-oss/enigma-go"
+	"github.com/qlik-oss/gopherciser/action"
+	"github.com/qlik-oss/gopherciser/enigmahandlers"
+	"github.com/qlik-oss/gopherciser/globals/constant"
+	"github.com/qlik-oss/gopherciser/logger"
+	"github.com/qlik-oss/gopherciser/senseobjdef"
+)
+
+type (
+	ObjectHandler interface {
+		Instance(id string) ObjectHandlerInstance
+	}
+
+	objectHandlerMap struct {
+		m         map[string]ObjectHandler
+		writeLock sync.Mutex
+	}
+)
+
+const (
+	maxNbrLines = 12
+	maxNbrTicks = 300
+)
+
+var (
+	GlobalObjectHandler objectHandlerMap
+)
+
+func init() {
+	// Register default object handlers
+	if err := GlobalObjectHandler.RegisterHandler("auto-chart", &AutoChartHandler{}, false); err != nil {
+		panic(err)
+	}
+}
+
+// RegisterHandler for object type, override existing handler with override flag
+func (objects *objectHandlerMap) RegisterHandler(objectType string, handler ObjectHandler, override bool) error {
+	objects.writeLock.Lock()
+	defer objects.writeLock.Unlock()
+
+	if objects.m == nil {
+		objects.m = make(map[string]ObjectHandler)
+	}
+
+	// Does a handler exit?
+	_, exists := objects.m[objectType]
+	if exists && !override {
+		return errors.New(fmt.Sprintf("object type<%s> already has a handler registered", objectType))
+	}
+
+	objects.m[objectType] = handler
+	return nil
+}
+
+// GetObjectHandler for objectType
+func (objects *objectHandlerMap) GetObjectHandler(objectType string) ObjectHandler {
+	handler, ok := objects.m[objectType]
+	if ok {
+		return handler
+	}
+	return &DefaultHandler{}
+}
+
+// GetAndAddObjectAsync get and add object to object handling
+func GetAndAddObjectAsync(sessionState *State, actionState *action.State, name, oType string) {
+	sessionState.QueueRequest(func(ctx context.Context) error {
+		sessionState.LogEntry.LogDebugf("object<%s> type<%s> found", name, oType)
+		sense := sessionState.Connection.Sense()
+
+		var genObj *enigma.GenericObject
+		getObject := func(ctx context.Context) error {
+			var err error
+			genObj, err = sense.CurrentApp.Doc.GetObject(ctx, name)
+			return err
+		}
+		if err := sessionState.SendRequest(actionState, getObject); err != nil {
+			return errors.Wrapf(err, "Failed go get object<%s>", name)
+		}
+
+		obj, err := sense.AddNewObject(genObj.Handle, enigmahandlers.ObjTypeSheetObject, name, genObj)
```

Comment:
I know it's old but maybe the name "ObjTypeSheetObject" isn't ideal

Reply:
Agree, didn't do this type of refactoring. Maybe `ObjTypeGenericObject` works better?

Reply:
Yeah I think that better describes it